### DISCUSSION TITLE
@wagmi/core V2: Checking last connection before Reconnect

### DISCUSSION THREAD

**Discussion by Nilbristi** [wevm/wagmi#3405] [General]
Hello Developers,

I am now migrating to @wagmi/core v2. After reading the migration document I came to now that autoConnect feature has been removed and the reconnecting behavior must be managed manually and is not related to the Wagmi Config. Use the [reconnect](https://wagmi.sh/core/api/actions/reconnect) action instead.

But I just can't understand how can I detect the last connector? I have multiple connectors like metamask, wallectConnect and Web3AuthConnector ("@web3auth/web3auth-wagmi-connector").

What I understand that I need the help of localstorage. But can't understand what will be condition to call reconnect function, on page refresh or reopen the site?

```
if(<condition to check>){
  await reconnect(this.config, {
        connectors: [<last connector>],
      }).then((success: any) => {
        console.log('\n==== wallet reconnected ====\n', success, '\n====');
      },
      (err: any) => {
        console.log('\n==== wallet reconnection issue ====\n', err, '\n====');
      })
 }
```

**Comment by tmm** [comment]
> But I just can't understand how can I detect the last connector?

You can omit the `connectors` parameter and Wagmi will [retrieve the `recentConnectorId` from `storage`](https://github.com/wevm/wagmi/blob/main/packages/core/src/actions/reconnect.ts#L46) or you can get it yourself with `const recentConnectorId = await config.storage?.getItem('recentConnectorId')`.

> But can't understand what will be condition to call reconnect function, on page refresh or reopen the site?

Depends on the framework you are using or how your app is set up.

**Reply by Nilbristi** [reply]
Thank you @tmm

**Comment by Nilbristi** [comment]
@tmm Can you please check this query https://github.com/wevm/wagmi/discussions/3414

**Reply by tmm** [reply]
I answer questions as I have time. Unfortunately, I can't answer all of them. @ mentioning me makes my notifications less useful so please stop.

**Reply by Nilbristi** [reply]
Extremely sorry
